compiler/src/dotty/tools/dotc/rewrites/Rewrites.scala

```diff
@@ -20,13 +20,16 @@ object Rewrites {
     private[Rewrites] val pbuf = new mutable.ListBuffer[Patch]()
 
     def addPatch(span: Span, replacement: String): Unit =
-      pbuf += Patch(span, replacement)
+      pbuf.indexWhere(p => p.span.start == span.start && p.span.end == span.end) match {
+        case i if i >= 0 => pbuf.update(i, Patch(span, replacement))
+        case _           => pbuf += Patch(span, replacement)
+      }
```

Comment:
I suspect this to be the root cause of https://github.com/lampepfl/dotty/issues/17187

```scala
object A:
    object B:
        def a = 2
```
When rewriting to no-indent, we want to add `  }\n` and then `}\n` at the exact same position. But this piece of code only keep the second patch, which leads to the incorrect:
```scala
object A {
    object B {
        def a = 2
}
```